framework/bundles/org.jboss.tools.rsp.secure/src/org/jboss/tools/rsp/secure/crypto/CryptoUtils.java

```diff
@@ -0,0 +1,113 @@
+package org.jboss.tools.rsp.secure.crypto;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.security.Security;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * A utility class that encrypts or decrypts a file.
+ * 
+ * @author www.codejava.net
+ *
+ */
+public class CryptoUtils {
+	private static final String ALGORITHM = "AES";
```

Comment:
The Downside

The main problem is that a class depending on a static method from a Utility Class has tight coupling. You are using a specific external dependency, not an abstraction. There is no way to switch that dependency under various circumstances. Usually, you would be able to provide a subclass of that dependency instead or another dependency implementing the interface. You lose that flexibility completely.
Testing

Since you have a hard dependency on the Util Class, you cannot easily provide a mock implementation. That makes testing rather difficult. You cannot isolate dependencies on Utility Classes and test just your class in isolation. Therefore you are testing all your static dependencies as well. Miško Hevery has a nice article about that - Static Methods are Death to Testability.

Static-method-only utility classes are basically the devil ;)